tests/phpunit/CRM/MembersOnlyEvent/Service/MembersOnlyEventAccessTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+use CRM_MembersOnlyEvent_Service_MembersOnlyEventAccess as MembersOnlyEventAccess;
+use CRM_MembersOnlyEvent_Test_Fabricator_MembersOnlyEvent as MembersOnlyEventFabricator;
+use CRM_MembersOnlyEvent_Test_Fabricator_Contact as ContactFabricator;
+use CRM_MembersOnlyEvent_Test_Fabricator_MembershipType as MembershipTypeFabricator;
+use CRM_MembersOnlyEvent_Test_Fabricator_Membership as MembershipFabricator;
+use CRM_MembersOnlyEvent_BAO_EventMembershipType as EventMembershipType;
+
+require_once __DIR__ . '/../../../BaseHeadlessTest.php';
+
+/**
+ * Class CRM_MembersOnlyEvent_BAO__MembersOnlyEventPriceFieldValue
+ *
+ * @group headless
+ */
+class CRM_MembersOnlyEvent_Service_MembersOnlyEventAccessTest extends BaseHeadlessTest {
+
+  /**
+   * @param int $value
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function setMembershipDurationCheckSettings($value) {
+    civicrm_api3('Setting', 'create', [
+      'membership_duration_check' => $value,
+    ]);
+  }
+
+  /**
+   * @return int
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function createLoggedInUser() {
+    $contactID = ContactFabricator::fabricate()['id'];
+
+    $session = CRM_Core_Session::singleton();
+    $session->set('userID', $contactID);
+    return $contactID;
+  }
+
+  /**
+   * Tests IsValidPath() happy path.
+   */
+  public function testIsValidPathHappyPath() {
```

Comment:
I think that "happy" and "unhappy" does not mean much in this context, I would even separate this test into two tests, maybe : 

testIfEventInfoPageIsValidPath()

testIfEventRegisterationPageIsValidPath()

testIfNonEventInfoOrRegistrationPageIsInvalidPath